src/index.ts

```diff
@@ -1,173 +1,231 @@
+import * as stringify_ from 'json-stringify-pretty-compact';
+import { Item, ScenegraphEvent, View } from 'vega-typings';
+import { isArray, isObject, isString } from 'vega-util';
+
+const stringify = (stringify_ as any).default || stringify_;
+
+export const DEFAULT_OPTIONS = {
+  /**
+   * X offset.
+   */
+  offsetX: 10,
+
+  /**
+   * Y offset.
+   */
+  offsetY: 10,
+
+  /**
+   * If of the tooltip element.
+   */
+  id: 'vg-tooltip-element',
+
+  /**
+   * The name of the theme. You can use the CSS class called [THEME]-theme to style the tooltips.
+   *
+   * There are two predefined themes: "light" (default) and "dark".
+   */
+  theme: 'light',
+};
+
+export type Options = typeof DEFAULT_OPTIONS;
+
+const STYLE = `
+.vg-tooltip {
+  visibility: hidden;
+  padding: 8px;
+  position: fixed;
+  z-index: 1000;
+  font-family: sans-serif;
+  font-size: 11px;
+  border-radius: 3px;
+  box-shadow: 2px 2px 4px rgba(0,0,0,0.1);
+
+  /* The default theme is the light theme. */
+  background-color: rgba(255, 255, 255, 0.95);
+  border: 1px solid #d9d9d9;
+  color: black;
+}
+.vg-tooltip.visible {
+  visibility: visible;
+}
+.vg-tooltip h2 {
+  margin-top: 0;
+  margin-bottom: 10px;
+  font-size: 13px;
+}
+.vg-tooltip table {
+  border-spacing: 0;
+}
+.vg-tooltip td {
+  overflow: hidden;
+  text-overflow: ellipsis;
+  padding-top: 2px;
+  padding-bottom: 2px;
+}
+.vg-tooltip td.key {
+  color: #808080;
+  max-width: 150px;
+  text-align: right;
+  padding-right: 4px;
+}
+.vg-tooltip td.value {
+  max-width: 200px;
+  text-align: left;
+}
+
+/* Dark and light color themes */
+.vg-tooltip.dark-theme {
+  background-color: rgba(32, 32, 32, 0.9);
+  border: 1px solid #f5f5f5;
+  color: white;
+}
+.vg-tooltip.dark-theme td.key {
+  color: #bfbfbf;
+}
 
-let tooltipPromise: number = undefined;
-let tooltipActive = false;
+.vg-tooltip.light-theme {
+  background-color: rgba(255, 255, 255, 0.95);
+  border: 1px solid #d9d9d9;
+  color: black;
+}
+.vg-tooltip.light-theme td.key {
+  color: #808080;
+}
+`;
 
 /**
- * Export API for Vega visualizations: vg.tooltip(vgView, options)
- * options can specify whether to show all fields or to show only custom fields
- * It can also provide custom title and format for fields
+ * The tooltip html element.
  */
-export function vega(vgView: VgView, options: Option = {showAllFields: true, isComposition: false}) {
-  start(vgView, copyOptions(options));
+let tooltipElement: HTMLElement;
 
-  return {
-    destroy: function () {
-      // remove event listeners
-      vgView.removeEventListener('mouseover.tooltipInit');
-      vgView.removeEventListener('mousemove.tooltipUpdate');
-      vgView.removeEventListener('mouseout.tooltipRemove');
+/**
+ * Initialize the tooltip element and style.
+ *
+ * @param options Tooltip options.
+ */
+function init(options: Options) {
+  if (tooltipElement) {
```

Comment:
What if init is called twice with different id ?   I would recommend instead to try to do  getElementById(options.id), and proceed if it doesn't exist.